src/slam_karto.cpp

```diff
@@ -1207,6 +1226,37 @@ SlamKarto::updateMap()
       scan->SetTime(source_scan->GetTime());
       scans.push_back(scan);
     }
+
+    // Populate the graph
```

Comment:
Does it make sense to do this only if there are listeners to the graph updates? We will be spending some cycles populating the updates, but they will be useless to the clients. Or general understanding is to first request the complete graph and then receive updates?

Reply:
That would work...except that there is a service call that assumes the `graph_msg_` is already populated.
I'm generally making sure that Aaron has whatever options he needs.
In my mind, you would need some sort of fallback plan if you subscribe to the "graph_updates" topic and miss one or more of the incremental updates. Or, as you say, you need to get the full graph in the very beginning and then apply incremental updates from that starting point.